options.go

```diff
@@ -107,17 +113,29 @@ func (vo *opts) filter(s stack.Stack) bool {
 	return false
 }
 
-func (vo *opts) retry(i int) bool {
-	if i >= vo.maxRetries {
-		return false
-	}
+func (vo *opts) newRetry() func() bool {
+	start := time.Now()
+	count := float64(-2)
 
-	d := time.Duration(int(time.Microsecond) << uint(i))
-	if d > vo.maxSleep {
-		d = vo.maxSleep
+	return func() bool {
+		count++
+		if count < 0 {
+			return true
+		}
+
+		if time.Since(start) >= vo.maxRetry {
+			return false
+		}
+
+		// simple backoff algorithm
+		d := time.Duration(math.Pow(2, count)) * time.Microsecond
```

Comment:
Instead of using `math.Pow`, you could instead use a state variable for the time to sleep, and multiply it by 2,

```
sleep := time.Microsecond
```

and then in the loop,
```
sleep = sleep * 2
if sleep >= vo.maxSleep {
  sleep = vo.maxSleep
}
```

simpler and more efficient